src/app/core/services/videos-list.service.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from '@angular/core';
+import { Observable, BehaviorSubject } from 'rxjs';
+
+import { Video } from '@core/models/video.model';
+import { LocalStorageService } from './local-store.service';
+import { NewVideoService } from './new-video.service';
+import demo from '@tools/demo';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class VideosListService {
+	private videos: Video[] = this.localStorageService.getLocalStorage();
+
+	private videosBehaviorSubject = new BehaviorSubject<Video[]>([]);
+	public videos$: Observable<Video[]> = this.videosBehaviorSubject.asObservable();
+
+	constructor(public newVideoService: NewVideoService, public localStorageService: LocalStorageService) {
+		this.videosBehaviorSubject.next(JSON.parse(window.localStorage.getItem('video-db')));
+	}
+
+	public adVideo(video: Video): void {
+		this.videos.push(video);
+
+		this.videosBehaviorSubject.next(this.videos);
+		this.localStorageService.saveToLocalStorage(this.videos);
```

Comment:
w kazdym sie to powtarza, mozna wyodrebnic do osobnej metody

Reply:
"a mozna to umiescic nad 15 linijką? tj wyciągnąć z local storage i po prostu wrzucic jako initial value do behaviorSubjecta?" próbowałem na kilka sposobów i zachowanie apki nie było poprawne

Reply:
nie wiem czemu tu, ale moj komentarz pod wlasciwym miejscem sie kurde nie wyswietla
w kazdym razie:
```
private videos: Video[] = JSON.parse(localStorage.getItem('videos') || '[]') as Video[];
private _video: BehaviorSubject<Video[]> = new BehaviorSubject<Video[]>(this.videos);
public videos$: Observable<Video[]> = this._video.asObservable();
```
myslalem o czyms takim, i u Seby tak jest